icu4c/source/test/intltest/restsnew.cpp

```diff
@@ -1343,5 +1354,82 @@ void NewResourceBundleTest::TestFilter() {
     }
 }
 
+#if U_ENABLE_TRACING
+
+static std::vector<std::string> gResourcePathsTraced;
+static std::vector<std::string> gDataFilesTraced;
+static std::vector<std::string> gResFilesTraced;
+
+static void U_CALLCONV traceData(
+        const void*,
+        int32_t fnNumber,
+        int32_t,
+        const char *,
+        va_list args) {
+
+    if (fnNumber == UTRACE_RESOURCE_PATH) {
+        va_arg(args, const char*); // type
+        va_arg(args, const char*); // file
+        const char* resourcePath = va_arg(args, const char*);
+        gResourcePathsTraced.push_back(resourcePath);
+    } else if (fnNumber == UTRACE_DATA_FILE_PATH) {
+        const char* filePath = va_arg(args, const char*);
+        gDataFilesTraced.push_back(filePath);
+    } else if (fnNumber == UTRACE_RES_FILE_PATH) {
+        const char* filePath = va_arg(args, const char*);
+        gResFilesTraced.push_back(filePath);
+    }
+}
+
+void NewResourceBundleTest::TestTrace() {
+    IcuTestErrorCode status(*this, "TestTrace");
+
+    const void* context;
+    utrace_setFunctions(context, nullptr, nullptr, traceData);
+    utrace_setLevel(UTRACE_VERBOSE);
+
+    {
+        LocalPointer<BreakIterator> brkitr(BreakIterator::createWordInstance("zh-CN", status));
+
+        assertEquals("Should touch expected resource paths",
+            { "/boundaries/word" },
+            gResourcePathsTraced);
+        assertEquals("Should touch expected data files",
+            { "icudt64l-brkitr/word.brk" },
```

Comment:
Please use uvernum.h to make this work across versions and endian/charset.

Reply:
Good catch.